client/transport/client.go

```diff
@@ -21,6 +21,8 @@ type Client interface {
 	Scheme() string
 	// TLSConfig returns any TLS configuration the client uses.
 	TLSConfig() *tls.Config
+
+	Close()
```

Comment:
Close should return an error so that it matches the `io.Closer` interface

`Close() error`